apps/src/aichat/redux/thunks/submitChatContents.ts

```diff
@@ -59,10 +62,13 @@ export const submitChatContents = createAsyncThunk(
       modelParameters,
       clientType,
       analyticsProperties,
+      userAddedContext,
     } = newUserMessageInput;
 
     // Clear any staged files if present (used with multimodal models)
     thunkAPI.dispatch(clearStagedFiles());
+    // Clear any user added context if present.
+    thunkAPI.dispatch(clearUserAddedContext());
```

Comment:
When a user changes levels, it seems like we should also clear user added context since the files are from the level's sources. 

For staged files, it doesn't matter seem to matter as much since they're uploaded by user? But maybe we want to clear staged files also when the level changes?

Reply:
good catch, I'll clear the user added context on level switch. I think it also makes sense to clear the staged files on level switch too, although I guess AI Chat would also hit this issue so I wonder if it was intentional or not. @edcodedotorg or @cnbrenci  do you know?

Reply:
nvm we were already clearing staged files on level change [here](https://github.com/code-dot-org/code-dot-org/blob/aad149acd9b8f8cd5501c28c1ff428aed932e12b/apps/src/aichat/views/ChatWorkspace.tsx#L154). Will update to also clear user added context!